### PR TITLE
Refine release notes for 8.5

### PR DESCRIPTION
**Pull request by lkasso**
Cleanup release notes for 8.5 to follow guidelines more closely.

### Reviewers:
Code owners, please let me know if these changes are acceptable.

### Guidelines for release notes:
* Set the context - what are we even talking about (don’t assume too much about what the user already knows)
* Express the problem or use case in the given context
* Explain how the new release addresses that problem or use case
* Provide build script snippet or screenshot illustrating the new feature, if applicable

### CODE REVIEW

**Review by tresat — APPROVED** [review]

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -102,6 +104,8 @@ versionCatalogs.named("libs").findLibrary("assertj-core").ifPresent { assertjCor
 }
 ```
 
+This was done to ensure that version catalogs are accessible during the configuration phase of a multi-module project, avoiding cumbersome workarounds.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:107]: @eskatos, is this an accurate statement?

Comment (lkasso) [reply subprojects/docs/src/docs/release/notes.md:107]: I want to make sure release notes don't just address the **what** but also the **why**. Why did we add this? How does it benefit our users?

Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:107]:
```suggestion
This was done to ensure that version catalogs are accessible via API in situations where generated accessors are not.
```

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -102,6 +104,8 @@ versionCatalogs.named("libs").findLibrary("assertj-core").ifPresent { assertjCor
 }
 ```
 
+This was done to ensure that version catalogs are accessible via API in situations where generated accessors are not.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:107]: @eskatos, I updated the statement but it still doesn't feel very impactful.

Comment (lkasso) [reply subprojects/docs/src/docs/release/notes.md:107]: I'm not a fan of this, I will delete and merge.

Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:107]:
```suggestion
```

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -102,6 +104,7 @@ versionCatalogs.named("libs").findLibrary("assertj-core").ifPresent { assertjCor
 }
 ```
 
+
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:107]:
```suggestion
```

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (eskatos) [reply subprojects/docs/src/docs/release/notes.md]: The new statement is more accurate. Feel free to omit though, I think it just paraphrases what's above.

**Pull request merged**